src/blockchain/rpc.rs

```diff
@@ -315,6 +339,10 @@ impl Blockchain for RpcBlockchain {
             debug!("adding utxo: {:?}", s);
             db.set_utxo(s)?;
         }
+        for s in spent_in_mempool {
+            debug!("adding utxo: {:?}", s);
+            db.set_utxo(&s)?;
+        }
```

Comment:
Feel like named variable would help here too..

```suggestion
        for utxo in spent_in_mempool {
            debug!("adding utxo: {:?}", utxo);
            db.set_utxo(&utxo)?;
        }
```